tests/unit/s2n_post_handshake_test.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include "error/s2n_errno.h"
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_key_update.h"
+#include "tls/s2n_post_handshake.h"
+#include "utils/s2n_safety.h"
+
+int s2n_key_update_write(struct s2n_blob *out);
+
+int main(int argc, char **argv)
+{
+
+    BEGIN_TEST();
+
+    /* This test checks that the post_handshake_recv function correctly processes a post_handshake keyupdate message. */
```

Comment:
I think this test is identical to the one below? If we just want to verify that post_handshake_recv clears any data given to it, let's just test it once. If you want to test that it correctly routes key update messages to key update though, you'll need one indicator to check that a key update message got where it needed to go. You could try checking the key_update_pending flag.